src/Data/List/Properties.agda

```diff
@@ -912,9 +910,9 @@ module _ {P : Pred A p} (P? : Decidable P) where
   length-partition : ∀ xs → (let (ys , zs) = partition P? xs) →
                      length ys ≤ length xs × length zs ≤ length xs
   length-partition []       = z≤n , z≤n
-  length-partition (x ∷ xs) with does (P? x) | length-partition xs
-  ...  | true  | rec = Prod.map s≤s m≤n⇒m≤1+n rec
-  ...  | false | rec = Prod.map m≤n⇒m≤1+n s≤s rec
+  length-partition (x ∷ xs) with does (P? x)
+  ...  | true  = Prod.map s≤s m≤n⇒m≤1+n (length-partition xs)
+  ...  | false = Prod.map m≤n⇒m≤1+n s≤s (length-partition xs)
```

Comment:
So this one is kind of interesting, when I originally wrote this code I put the second `with` in in order to reduce code duplication. Can we somehow maintain it so we only have to write `length-partition xs` once without the `with`?

Reply:
I think that is/would be possible with an irrefutable with pattern. In the first instance I had been trying to avoid those, but they are at least more notationally economical than the previous versions. What about the computational behaviour at typechecking time though?

Reply:
But note: this kind of idiomatic use if with is very common... and I wasn't aiming for a systematic trawl through all of them, at this stage at least!

Reply:
So: I see (at least) two possible solutions:
* a `where clause`, but I think they don't (necessarily) scope correctly over all the branches of a `with`, so I don't think works in this case (but see below)
* an irrefutable ` with rec ← <recursive call>` idiom, which has to *precede* the (Boolean test)
* rewrite to the Boolean with `if_then_else_`, and then use a `where`/`let` clause for the recursive (or not at all, if not actually needed; some of these could simply be inlined)

I think the request for discussion is that I don;t know which of these alternatives is actually preferable (I raised the `if_then_else_` issue on issue #1937 itself). Which do you prefer?

Reply:
Yes, I'm interested in how we resolve this problem in general. Current solution seems okay.

Reply:
We should document our "use cases" and post them on the main Agda issue tracker. I think this particular situation doesn't have a nice solution in now-agda and may well be a simple extension.